[enterprise-4.14] OSDOCS-6677: add RN entry (#65889)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

For more information, see xref:../installing/install_config/installing-customizing.adoc#installation-special-config-encryption-threshold_installing-customizing[Configuring an encryption threshold] and xref:../installing/install_config/installing-customizing.adoc#installation-special-config-storage-procedure_installing-customizing[Configuring disk encryption and mirroring].

[id="ocp-4-14-new-cert-process"]
==== Certificates are now handled by the Machine Config Daemon

In previous {product-title} versions, the MCO read and handled certificates directly from machine configuration files. This led to rotation issues and created unwanted situations, like when certificates were stuck behind a paused machine config pool.

> With this release, certificates are no longer templated from bootstrap into machine configuration files. Instead, they are put directly onto disk using controller configs and handled by the Machine Config Daemon (MCD). The certs are then visible using the Controller Config.

Reviewer: ```suggestion
With this release, certificates are no longer templated from bootstrap into machine configuration files. Instead, they are put directly into the ignition object, written onto disk using the controller config, and handled by the Machine Config Daemon (MCD) during regular cluster operation. The certs are then visible using the Controller Config.
```

> With this release, certificates are no longer templated from bootstrap into machine configuration files. Instead, they are put directly into the Ignition object, written onto disk using the controller config, and handled by the Machine Config Daemon (MCD) during regular cluster operation. The certs are then visible using the Controller Config.

Reviewer 2: ```suggestion
With this release, certificates are no longer templated from bootstrap into machine configuration files. Instead, they are put directly into the Ignition object, written onto a disk using the controller config, and handled by the Machine Config Daemon (MCD) during regular cluster operation. The certs are then visible using the Controller Config.
```

Reviewer 2: I think an a/the is needed before "disk". Also, I was wondering about Controller Config needing to be capitalized? In the docs, I see config controller, lowercase. I'm unfamiliar with this, so I just wanted to ask. 

Also is Ignition an API object that needs backticks?

Author: Ignition doesn't need backticks, I think you're right that controller config should either be lowercase or it should be in backticks e.g. `ControllerConfig` resource...this is a new area for me as well 😄

> The MCD handles the following certificates:

Reviewer: The Controller holds the certificate data but the MCD writes the data to disk so it is kind of a partnership. I guess we can be more specific here and say the Data is stored in and retrieved from the controller config and written to disk by the Daemon.

> The MCD handles the following certificates:

Author: I've added a couple of sentences about the MCC and image registry certificates

* `/etc/kubernetes/kubelet-ca.crt`
* `/etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem`
* `/etc/pki/ca-trust/source/anchors/openshift-config-user-ca-bundle.crt`

This means that certificates are not bound by the machine config pool status and are more timely in their rotation.

> Any CAs stored in machine configuration files have been removed, and the templated files found during cluster installation no longer exist.

Reviewer: ```suggestion
The above listed CAs stored in machine configuration files have been removed, and the templated files found during cluster installation no longer exist.
```

> The above listed CAs stored in machine configuration files have been removed, and the templated files found during cluster installation no longer exist.

Reviewer 2: ```suggestion
The previously listed CAs stored in machine configuration files are removed, and the templated files found during cluster installation no longer exist.
```

Reviewer 2: IBMSG adivises caution when using above and suggests using previously or preceding instead. 
https://www.ibm.com/docs/en/ibm-style?topic=word-usage#a

Also, not sure if this makes sense here, but I was thinking active voice is better.

Author: totally makes sense 🙂
